packages/core/src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -141,14 +151,46 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }
 
-    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    function handleChange(event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
       setInputValue(event.target.value);
+
+      if (multiline) {
+        resizeTextarea();
+      }
     }
 
     const toggleKeyboardEditMode = useKeyboardButtonPressedFunc(toggleEditMode);
 
     function focus() {
       inputRef.current?.focus?.();
+
+      if (multiline) {
+        calculateTextareaHeightAttrs();
+      }
+    }
+
+    /* Dynamically resizes the textarea to fit its content */
+    function resizeTextarea() {
+      if (inputRef.current) {
+        // Temporarily set the height to "auto" to accurately measure the scroll height of the content inside the textarea.
+        setInputHeight("auto");
+
+        requestAnimationFrame(() => {
+          const textarea = inputRef.current as HTMLTextAreaElement;
+
+          if (!textarea) {
+            return;
+          }
+
+          // Ensure we at least have 1 line
+          setInputHeight(
+            Math.max(
+              textarea.scrollHeight + textareaBorderBoxSizing.current,
+              textareaLineHeight.current + textareaBorderBoxSizing.current
```

Comment:
Just one thought, looks like the only difference in the dimensions is 2px, and there's no difference in paddings/line heights etc between the different variants, so looks like doing just:
`setInputHeight(textarea.scrollHeight + 2);` works exactly the same and it can save the getComputeStyle and the other calculations in `calculateTextareaHeightAttrs` (which already have some default static numbers). Giving it `auto` in the beginning will already make it take the the currect box size. WDYT?

Reply:
actually the question gets back to you :)
technically its only 2px because the border-top/bottom-width is 1px (or 2px overall)
border-box should have included the border as well as the padding, but it doesn't

do we want to just rely on always having this border/padding setting and that the user cannot add anything on their own using overwriting css?
@talkor

Reply:
Well it is border-box, it's just that `scrollHeight` is based purely on the content and padding of the element's scrollable content, without the borders. We don't want users to override CSS as it can make design inconsistent so I wouldn't take it into consideration, generally speaking we support the variants we officially provide so I think it's ok :)